src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -79,6 +79,9 @@ public function request(string $method, string $url, array $options = []): Respo
             if (str_starts_with($options['bindto'], 'host!')) {
                 $options['bindto'] = substr($options['bindto'], 5);
             }
+            if ((\PHP_VERSION_ID < 80223 || 80300 <= \PHP_VERSION_ID && 80311 < \PHP_VERSION_ID) && '\\' === \DIRECTORY_SEPARATOR && '[' === $options['bindto'][0]) {
+                $options['bindto'] = preg_replace('{^\[[^\]]++\]}', '[$0]', $options['bindto']);
```

Comment:
This fixes the failure on Windows. I verified that on Windows, before PHP 8.2.23, `bindto` needs double brackets for binding to IPv6 addresses (`[[::1]]:0`).

I looked at the changelog of PHP 8.2.23 and the only PR that looks related is
https://github.com/php/php-src/pull/15068

I also compiled PHP 8.2.22 on my Ubuntu and the issue doesn't show up, while it does on Windows with the same version.